.github/workflows/qac.yml

```diff
@@ -0,0 +1,113 @@
+name: Code Analysis
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+
+  coverage:
+    name: Run Coverage 
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: 8.x
+
+      - name: Cache .NET packages
+        uses: actions/cache@v4
+        with:
+          path: ~/.nuget/packages
+          key: ${{ runner.os }}-dotnet-${{ hashFiles('**/*.csproj') }}
+          restore-keys: |
+            ${{ runner.os }}-dotnet-
+
+      - name: Restore dependencies
+        run: |
+          dotnet restore
+          dotnet tool restore
+
+      - name: Run coverage code 
+        run: |
+          dotnet tool install --global coverlet.console
+          dotnet build --no-incremental
+          dotnet test --results-directory ./coverage-report --collect:"XPlat Code Coverage"
```

Comment:
Do we need this `dotnet test` if we already use coverlet cli ?

Reply:
Merged dotnet test with covelet cli